openml-h2o/src/main/java/com/feedzai/openml/h2o/algos/AbstractH2OAlgoUtils.java

```diff
@@ -0,0 +1,169 @@
+     */
+    protected abstract T getEmptyParams();
+
+    /**
+     * Template method to parse H2O algorithm params.
+     *
+     * @param trainingFrame The dataset to be used.
+     * @param params        The raw training params.
+     * @param randomSeed    The source of randomness.
+     * @param datasetSchema The dataset schema.
+     * @return The modified version of the given {@code h2oParams}.
+     */
+    protected final T parseParams(final Frame trainingFrame,
+                               final Map<String, String> params,
+                               final long randomSeed,
+                               final DatasetSchema datasetSchema) {
+        final T baseParams = commonParams(trainingFrame, datasetSchema);
+        return parseSpecificParams(baseParams, params, randomSeed);
+    }
+
+    /**
+     * Auxiliary method to setup the common training params to all algorithms/models.
+     *
+     * @param trainingFrame The dataset to be used.
+     * @param datasetSchema The dataset schema.
+     * @return A modified version of the provided params object.
+     */
+    protected abstract T commonParams(Frame trainingFrame, DatasetSchema datasetSchema);
+
+    /**
+     * Abstract method that creates and returns the Model that this class refers to.
+     * @param h2oParams concrete implementation of {@link ModelParametersSchemaV3 algorithm params}.
+     * @return concrete implementation of {@link ModelBuilder H20 Model Builder} created from the h2oParams.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public abstract M getModel(T h2oParams);
+
+    /**
+     * Validates the h20 model parameters by building a model and validating it.
+     * @param paramsToValidate H20 parameters to validate.
+     * @param randomSeed       The source of randomness.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public void validateParams(Map<String, String> paramsToValidate, final long randomSeed){
+        final T baseParams = getEmptyParams();
+        final T updatedParams = parseSpecificParams(baseParams, paramsToValidate, randomSeed);
+        M modelBuilder = getModel(updatedParams);
+        validateModel(modelBuilder);
+    }
+
+    /**
+     * Trains the H20 model.
+     * @param trainingFrame The dataset to be used.
+     * @param params        The raw training params.
+     * @param randomSeed    The source of randomness.
+     * @param schema        The dataset schema.
+     * @return Job resulting from he model training.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public Job train(final Frame trainingFrame,
+                     final Map<String, String> params,
+                     final long randomSeed,
+                     final DatasetSchema schema){
+        M model = getModel(trainingFrame, params, randomSeed, schema);
+        return model.trainModel();
+
+    }
+
+    /**
+     * Method that returns creates the model in a training scenario.
```

Comment:
Improve the description. I suggest something like `Gets the model for training scenario`